node/src/Pos/Explorer/Txp/Global.hs

```diff
@@ -52,14 +53,16 @@ extraOps (ExplorerExtra em (HM.toList -> histories) balances) =
     map (uncurry GS.PutAddrBalance) (MM.insertions balances) ++
     map GS.DelAddrBalance (MM.deletions balances)
 
--- CSE-203 FIXME Current time is used as timestamp, are you serious?
 applyBlund
     :: (MonadSlots ctx m, EGlobalApplyToilMode ctx m)
     => TxpBlund
     -> m ()
 applyBlund blund = do
-    curTime <- currentTimeSlotting
-    uncurry (eApplyToil curTime) $ blundToAuxNUndoWHash blund
+    -- First get the current @SlotId@ so we can calculate the time.
+    -- Then get when that @SlotId@ started and use that as a time for @Tx@.
+    mTxTimestamp <- getCurrentSlotBlocking >>= getSlotStart
```

Comment:
No, it's wrong. You should take a slot from `TxpBlund`.
```
-- [CSL-1156] Maybe find better approach (at least wrap into normal types).
type TxpBlock = Either (Some IsGenesisHeader) (Some IsMainHeader, TxPayload)
type TxpBlund = (TxpBlock, TxpUndo)
```

`TxpBlund` has `TxpBlock` inside. If it's `Left`, it's a genesis block which doesn't contain transactions. It doesn't have a slot, only epoch, but you can use e. g. `SlotId epoch minBound`. If it's `Right`, you can use `headerSlotL` lens.

Reply:
I didn't even know that a transaction has a slot :grin: 
Ok, thanks, will fix.

Reply:
No, not transaction, it's a block.
`TxpBlock` is just like `Block`, but contains reduced set of information (only what is needed for tx processing).